oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2RefreshTokenAuthenticationProvider.java

```diff
@@ -171,7 +173,20 @@ public Authentication authenticate(Authentication authentication) throws Authent
 
 		OAuth2RefreshToken currentRefreshToken = refreshToken.getToken();
 		if (!tokenSettings.reuseRefreshTokens()) {
-			currentRefreshToken = generateRefreshToken(tokenSettings.refreshTokenTimeToLive());
+			Duration refreshTokenTimeToLive = tokenSettings.refreshTokenTimeToLive();
+			boolean isPublicClient = !StringUtils.hasText(registeredClient.getClientSecret());
```

Comment:
Also, we can take out `!StringUtils.hasText(registeredClient.getClientSecret())` to another separate method. Thus, we can not use an additional variable `isPublicClient`.